### PR TITLE
Added module code

### PR DESCRIPTION
**Pull request by jekuaitk**
Adds the OS2Forms GetOrganized module (moved from https://github.com/itk-dev/os2forms_selvbetjening/tree/develop/web/modules/custom/os2forms_get_organized).

### CODE REVIEW

File: src/Form/SettingsForm.php [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Drupal\os2forms_get_organized\Helper\Settings;
+use ItkDev\GetOrganized\Client;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\OptionsResolver\Exception\ExceptionInterface as OptionsResolverException;
+
+/**
+ * GetOrganized settings form.
+ */
+final class SettingsForm extends FormBase {
```
Comment (stankut) [inline src/Form/SettingsForm.php]: what is the reason for making this class final?

File: src/Helper/Settings.php [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+use Drupal\Core\KeyValueStore\KeyValueFactoryInterface;
+use Drupal\Core\KeyValueStore\KeyValueStoreInterface;
+use Drupal\os2forms_get_organized\Exception\InvalidSettingException;
+use Drupal\os2forms_get_organized\Form\SettingsForm;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * General settings for os2forms_get_organized.
+ */
+final class Settings {
```
Comment (stankut) [inline src/Helper/Settings.php]: is there any specific idea behind introducing a custom class/way of string the Settings?

It seems like classic Config would do just fine here, and it does not require extra classes. Unless i am missing some idea behind it

quick tutorial: https://www.drupal.org/docs/drupal-apis/configuration-api/working-with-configuration-forms

File: src/Plugin/AdvancedQueue/JobType/ArchiveDocument.php [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+/**
+ * Archive document job.
+ *
+ * @AdvancedQueueJobType(
+ *   id = "Drupal\os2forms_get_organized\Plugin\AdvancedQueue\JobType\ArchiveDocument",
+ *   label = @Translation("Archive document in GetOrganized"),
+ * )
+ */
+final class ArchiveDocument extends JobTypeBase implements ContainerFactoryPluginInterface {
```
Comment (stankut) [inline src/Plugin/AdvancedQueue/JobType/ArchiveDocument.php]: i see that , the making the custom classes final is just a matter of style. But IMO that is redundant, perhaps

File: src/Exception/AttachmentElementNotFoundException.php [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+namespace Drupal\os2forms_get_organized\Exception;
+
+/**
+ * Attachment element not found Exception.
+ */
+class AttachmentElementNotFoundException extends \Exception {
```
Comment (stankut) [inline src/Exception/AttachmentElementNotFoundException.php]: didn't find it being used. Is it needed?

File: src/Helper/ArchiveHelper.php [hunk 5]
```diff
@@ -0,0 +1,441 @@
...
+namespace Drupal\os2forms_get_organized\Helper;
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\os2forms_get_organized\Exception\ArchivingMethodException;
+use Drupal\os2forms_get_organized\Exception\CitizenArchivingException;
+use Drupal\os2forms_get_organized\Exception\GetOrganizedCaseIdException;
+use Drupal\webform\Entity\WebformSubmission;
+use Drupal\webform_entity_print_attachment\Element\WebformEntityPrintAttachment;
+use ItkDev\GetOrganized\Client;
```
Comment (stankut) [inline src/Helper/ArchiveHelper.php:12]: is that an unmentioned dependency? Was able to find it anywhere, same goes for all ItkDev\GetOrganized classes

File: os2forms_get_organized.info.yml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+name: 'OS2Forms GetOrganized'
+type: module
+description: 'GetOrganized integration'
+package: OS2Forms
+core_version_requirement: ^9
+dependencies:
+  - drupal:webform
+  - drupal:advancedqueue
```
Comment (stankut) [inline os2forms_get_organized.info.yml:8]: missing dependency: webform_entity_print_attachment

**Review by stankut — CHANGES_REQUESTED** [review]

File: os2forms_get_organized.info.yml (hunk 6 above)
Comment (jekuaitk) [reply os2forms_get_organized.info.yml:8]: Added dependency in eb82f746b7a5587ff3a6ad31b2d12276ac261ec6.

File: src/Helper/ArchiveHelper.php (hunk 5 above)
Comment (jekuaitk) [reply src/Helper/ArchiveHelper.php:12]: From my understanding non drupal module package [dependencies](https://www.drupal.org/docs/develop/creating-modules/let-drupal-know-about-your-module-with-an-infoyml-file#s-complete-example) are not included in `os2forms_get_organized.info.yml`, but instead just required through `composer.json`.

File: src/Exception/AttachmentElementNotFoundException.php (hunk 4 above)
Comment (jekuaitk) [reply src/Exception/AttachmentElementNotFoundException.php]: Good catch. Removed in eb82f746b7a5587ff3a6ad31b2d12276ac261ec6.

File: src/Plugin/AdvancedQueue/JobType/ArchiveDocument.php (hunk 3 above)
Comment (jekuaitk) [reply src/Plugin/AdvancedQueue/JobType/ArchiveDocument.php]: It is indeed redundant. Changed in eb82f746b7a5587ff3a6ad31b2d12276ac261ec6.

File: src/Form/SettingsForm.php (hunk 1 above)
Comment (jekuaitk) [reply src/Form/SettingsForm.php]: It is indeed redundant. Removed in eb82f746b7a5587ff3a6ad31b2d12276ac261ec6.

File: src/Helper/Settings.php (hunk 2 above)
Comment (jekuaitk) [reply src/Helper/Settings.php]: I did consider doing this via config, but ultimately decided against it. In my opinion doing it the config way just gives us the hassle of having to tread carefully when config exporting etc.

File: src/Helper/ArchiveHelper.php (hunk 5 above)
Comment (stankut) [reply src/Helper/ArchiveHelper.php:12]: right, i didn't notice it in composer.json

File: src/Form/SettingsForm.php (hunk 1 above)
Comment (stankut) [reply src/Form/SettingsForm.php]: i don't see it removed in https://github.com/OS2Forms/os2forms_get_organized/commit/eb82f746b7a5587ff3a6ad31b2d12276ac261ec6

Comment (stankut) [reply src/Form/SettingsForm.php]: great, will you also remove it for `class GetOrganizedWebformHandler` ?

File: src/Helper/Settings.php (hunk 2 above)
Comment (stankut) [reply src/Helper/Settings.php]: yes, i see your point, 
however other contrib modules are using that as simple config object without any second thoughts: https://git.drupalcode.org/project/smtp/-/blob/8.x-1.x/src/Form/SMTPConfigForm.php#L107

i will leave it up to you.

File: src/Form/SettingsForm.php (hunk 1 above)
Comment (jekuaitk) [reply src/Form/SettingsForm.php]: My bad - they are now removed.

**Review by stankut — APPROVED** [review]

**Pull request merged**